receiver/bigipreceiver/scraper.go

```diff
@@ -0,0 +1,299 @@
+	cfg      *Config
+	settings component.TelemetrySettings
+	mb       *metadata.MetricsBuilder
+}
+
+// newScraper creates an initialized bigipScraper
+func newScraper(logger *zap.Logger, cfg *Config, settings component.TelemetrySettings) *bigipScraper {
+	return &bigipScraper{
+		logger:   logger,
+		cfg:      cfg,
+		settings: settings,
+		mb:       metadata.NewMetricsBuilder(cfg.Metrics),
+	}
+}
+
+// start initializes a new big-ip client for the scraper
+func (s *bigipScraper) start(_ context.Context, host component.Host) (err error) {
+	s.client, err = newClient(s.cfg, host, s.settings, s.logger)
+	return
+}
+
+// scrape collects and creates OTEL metrics from a Big-IP environment
+func (s *bigipScraper) scrape(ctx context.Context) (pmetric.Metrics, error) {
+	now := pcommon.NewTimestampFromTime(time.Now())
+
+	// validate we don't attempt to scrape without initializing the client
+	if s.client == nil {
+		return pmetric.NewMetrics(), errClientNotInit
+	}
+
+	collectedMetrics := false
+
+	// initialize auth token
+	err := s.client.GetNewToken(ctx)
+	if err != nil {
+		return pmetric.NewMetrics(), err
+	}
+
+	var scrapeErrors scrapererror.ScrapeErrors
+	// scrape metrics for virtual servers
+	virtualServers, err := s.client.GetVirtualServers(ctx)
+	if err != nil {
+		scrapeErrors.AddPartial(0, err)
+		s.logger.Warn("Failed to scrape virtual server metrics", zap.Error(err))
+	} else {
+		collectedMetrics = true
+		for key := range virtualServers.Entries {
+			virtualServerStats := virtualServers.Entries[key]
+			s.collectVirtualServers(&virtualServerStats, now)
+		}
+	}
+
+	// scrape metrics for pools
+	pools, err := s.client.GetPools(ctx)
+	if err != nil {
+		scrapeErrors.AddPartial(0, err)
+		s.logger.Warn("Failed to scrape pool metrics", zap.Error(err))
+	} else {
+		collectedMetrics = true
+		for key := range pools.Entries {
+			poolStats := pools.Entries[key]
+			s.collectPools(&poolStats, now)
+		}
+	}
+
+	// scrape metrics for pool members
+	poolMembers, err := s.client.GetPoolMembers(ctx, pools)
+	if err == errCollectedNoPoolMembers {
+		scrapeErrors.AddPartial(0, err)
+		s.logger.Warn("Failed to scrape pool member metrics", zap.Error(err))
+	} else {
+		if err != nil {
+			scrapeErrors.AddPartial(0, err)
+			s.logger.Warn("Failed to scrape some pool member metrics", zap.Error(err))
+		}
+
+		collectedMetrics = true
+		for key := range poolMembers.Entries {
+			poolMemberStats := poolMembers.Entries[key]
+			s.collectPoolMembers(&poolMemberStats, now)
+		}
+	}
+
+	// scrape metrics for nodes
+	nodes, err := s.client.GetNodes(ctx)
+	if err != nil {
+		scrapeErrors.AddPartial(0, err)
+		s.logger.Warn("Failed to scrape node metrics", zap.Error(err))
+	} else {
+		collectedMetrics = true
+		for key := range nodes.Entries {
+			nodeStats := nodes.Entries[key]
+			s.collectNodes(&nodeStats, now)
+		}
+	}
+
+	if !collectedMetrics {
+		return pmetric.NewMetrics(), errScrapedNoMetrics
+	}
+
+	return s.mb.Emit(), scrapeErrors.Combine()
```

Comment:
Are there any metrics not associated with a resource? If not, I think this and `collectedMetrics` are unnecessary.